mysql-test/suite/rocksdb/t/use_direct_reads_writes.test

```diff
@@ -5,43 +5,37 @@
 # caused an assertion in RocksDB.  Now it should not be allowed and the
 # server will not start with that configuration
 
-# Write file to make mysql-test-run.pl expect the "crash", but don't restart
-# the server until it is told to
---let $_server_id= `SELECT @@server_id`
---let $_expect_file_name= $MYSQLTEST_VARDIR/tmp/mysqld.$_server_id.expect
---exec echo "wait" >$_expect_file_name
-shutdown_server 10;
+--let SEARCH_FILE=$MYSQLTEST_VARDIR/tmp/rocksdb.use_direct_reads_writes.err
 
-# Clear the log
---exec echo "" >$MYSQLTEST_VARDIR/log/mysqld.1.err
+# Expect server crash on startup with invalid combination of arguments
+--source include/expect_crash.inc
+shutdown_server 10;
 
 # Attempt to restart the server with invalid options
---exec echo "restart:--rocksdb_use_direct_reads=1 --rocksdb_allow_mmap_reads=1" >$_expect_file_name
---sleep 0.1  # Wait 100ms - that is how long the sleep is in check_expected_crash_and_restart
---exec echo "restart:" >$_expect_file_name
-
-# Cleanup
---enable_reconnect
---source include/wait_until_connected_again.inc
---disable_reconnect
+--let $restart_parameters=restart:--log-error=$SEARCH_FILE --rocksdb_use_direct_reads=1 --rocksdb_allow_mmap_reads=1
+--replace_result $SEARCH_FILE LOG_FILE
+--source include/start_mysqld.inc
```

Comment:
Yeah, this is bizzarre...it seems that trying to use '--exec $MYSQLD' and '--exec $MYSQLD_CMD' with any kind of plugin load just hangs, no output log files, nothing. There must be some issue with the way the new shell is being created. I even tried something as simple as --plugin_load=$ROCKSDB_DIR/$ROCKSDB --plugin_load_add=$ROCKSDB_DIR/$ROCKSDB and it just doesn't want to launch the process. Echoing out the -exec just shows

> /ssd/percona/myrocks/percona-server-5.7/sql/mysqld --defaults-group-suffix=.1 --defaults-file=/ssd/percona/myrocks/percona-server-5.7/mysql-test/var/my.cnf --loose-tokudb-cache-size=512M --log-syslog=0 --log-error=/ssd/percona/myrocks/percona-server-5.7/mysql-test/var/tmp/rocksdb.use_direct_reads_writes.err --plugin_load_add=/ssd/percona/myrocks/percona-server-5.7/storage/rocksdb/ha_rocksdb.so --rocksdb_use_direct_reads=1 --rocksdb_allow_mmap_reads=1

Which looks fine to me :-\



This has got to be something totally stupid and obvious that I am missing.